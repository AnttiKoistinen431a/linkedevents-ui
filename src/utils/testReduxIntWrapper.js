import { Provider } from 'react-redux'
import React from 'react'
import { IntlProvider } from 'react-intl'
import translations from '../i18n'

export default (store, component, locale = 'fi') => {
  return (
    <Provider store={store}>
      <IntlProvider locale={locale} messages={translations[locale]}>
        {component}
      </IntlProvider>
    </Provider>
  )
}
