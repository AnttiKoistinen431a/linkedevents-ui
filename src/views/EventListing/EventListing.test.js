import configureStore from 'redux-mock-store'
import React from 'react'
import thunk from 'redux-thunk'
import {shallow} from 'enzyme'
import renderer from 'react-test-renderer'

import testReduxIntWrapper from '../../../__mocks__/testReduxIntWrapper'
import ConnectedEventListing, {EventListing} from './index'
import {mockUserEvents} from '__mocks__/mockData';
import {mockCurrentTime, resetMockDate} from '../../../__mocks__/testMocks';

jest.mock('../../utils/events', () => ({
    EventQueryParams: class Foo {
        setPublisher() {}
        setSort() {}
    },
    fetchEvents: () => ({
        then: () => ({finally: () => true}),
        finally: () => true,
    }),
}))

const mockStore = configureStore([thunk])
const initialStore = {
    subEvents: {
        bySuperEventId: {},
        fetchingFromSuperId: '',
    },
    userEvents: {
        isFetching: false,
        fetchComplete: true,
        items: mockUserEvents,
        error: null,
        sortBy: 'last_modified_time',
        sortOrder: 'desc',
        count: 738,
        paginationPage: 0,
    },
    user: {
        id: 'testuser',
        username: 'fooUser',
        provider: 'helsinki',
    },
    events: {
        apiErrorMsg: null,
        isFetching: false,
        fetchComplete: false,
        items: [],
        eventError: null,
        eventsError: null,
    },
    app: {
        flashMsg: null,
        confirmAction: null,
    },
}

describe('EventListing Snapshot', () => {
    let store;

    beforeEach(() => {
        mockCurrentTime('2018-11-10T12:00:00z')
    })

    afterEach(() => {
        resetMockDate()
    })

    it('should render view by default', () => {
        const componentProps = {
            login: jest.fn(),
            events: {},
            user: {},
        };
        const wrapper = shallow(<EventListing {...componentProps} />);
        expect(wrapper).toMatchSnapshot();
    })

    it('should render view correctly', () => {
        store = mockStore(initialStore)
        const componentProps = {
            login: jest.fn(),
        } // Props which are added to component
        const componentToTest = <ConnectedEventListing {...componentProps} />
        const wrapper = renderer.create(testReduxIntWrapper(store, componentToTest))

        expect(wrapper).toMatchSnapshot()
    })
})
