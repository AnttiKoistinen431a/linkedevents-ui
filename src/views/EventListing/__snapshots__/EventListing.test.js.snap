// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventListing Snapshot should render view by default 1`] = `
<div
  className="container"
>
  <h1>
    <FormattedMessage
      id="events-management"
      values={Object {}}
    />
  </h1>
  <p>
    <FormattedMessage
      id="events-management-description"
      values={Object {}}
    />
  </p>
  <Connect(FilterableEventTable)
    apiErrorMsg=""
    user={Object {}}
  />
</div>
`;

exports[`EventListing Snapshot should render view correctly 1`] = `
<div
  className="container"
>
  <h1>
    <span>
      Tapahtumien hallinta
    </span>
  </h1>
  <p>
    <span>
      Selaa ja muokkaa oman organisaatiosi julkisia tapahtumia sekä luonnoksia. Voit lisätä uuden tapahtuman yläpalkista.
    </span>
  </p>
  <table
    className="MuiTable-root-1 event-table"
  >
    <thead>
      <tr
        className="MuiTableRow-root-2 MuiTableRow-head-3"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeHead-10 WithStyles-TableCell--typeHead-7"
        >
          <span
            className="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span>
              Otsikko
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-23 MuiTableSortLabel-icon-18 MuiTableSortLabel-desc-19"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeHead-10 WithStyles-TableCell--typeHead-7"
        >
          <span
            className="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span>
              Tapahtuma alkaa
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-23 MuiTableSortLabel-icon-18 MuiTableSortLabel-desc-19"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeHead-10 WithStyles-TableCell--typeHead-7"
        >
          <span
            className="MuiButtonBase-root-21 MuiTableSortLabel-root-16"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span>
              Tapahtuma päättyy
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-23 MuiTableSortLabel-icon-18 MuiTableSortLabel-desc-19"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeHead-10 WithStyles-TableCell--typeHead-7"
        >
          <span
            className="MuiButtonBase-root-21 MuiTableSortLabel-root-16 MuiTableSortLabel-active-17"
            disabled={undefined}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="button"
            tabIndex={0}
          >
            <span>
              Muokattu viimeksi
            </span>
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-23 MuiTableSortLabel-icon-18 MuiTableSortLabel-desc-19"
              color={undefined}
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z"
              />
            </svg>
          </span>
        </th>
      </tr>
    </thead>
    <tbody>
      <tr
        className="MuiTableRow-root-2"
        onClick={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
          style={
            Object {
              "fontWeight": "normal",
              "paddingLeft": "24px",
            }
          }
        >
          <a
            href="/event/undefined"
            onClick={[Function]}
          >
            TestEvent1
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            Jan 31, 2018
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            Jan 31, 2018
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            10 kuukautta sitten
          </span>
        </td>
      </tr>
    </tbody>
    <tbody>
      <tr
        className="MuiTableRow-root-2"
        onClick={null}
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
          style={
            Object {
              "fontWeight": "normal",
              "paddingLeft": "24px",
            }
          }
        >
          <a
            href="/event/undefined"
            onClick={[Function]}
          >
            Pomppu
          </a>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            Feb 2, 2018
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            Feb 2, 2018
          </span>
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root-8 MuiTableCell-paddingDefault-13 MuiTableCell-typeBody-11 WithStyles-TableCell--typeBody-30"
        >
          <span>
            10 kuukautta sitten
          </span>
        </td>
      </tr>
    </tbody>
    <tfoot>
      <tr
        className="MuiTableRow-root-31 MuiTableRow-footer-33"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-45 MuiTableCell-paddingDefault-50 MuiTableCell-typeFooter-49 MuiTablePagination-root-36"
          colSpan={1000}
        >
          <div
            className="MuiToolbar-root-53 MuiToolbar-gutters-54 MuiTablePagination-toolbar-37"
          >
            <div
              className="MuiTablePagination-spacer-38"
            />
            <span
              className="MuiTypography-root-55 MuiTypography-caption-65 MuiTablePagination-caption-39"
            >
              1-100 / 738
            </span>
            <div
              className="MuiTablePaginationActions-root-79"
            >
              <button
                className="MuiButtonBase-root-89 MuiButtonBase-disabled-90 MuiIconButton-root-80 MuiIconButton-disabled-85"
                disabled={true}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role={undefined}
                tabIndex={-1}
                type="button"
              >
                <span
                  className="MuiIconButton-label-86"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-91 MuiIconButton-icon-87"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                    />
                  </svg>
                </span>
              </button>
              <button
                className="MuiButtonBase-root-89 MuiIconButton-root-80"
                disabled={false}
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                role={undefined}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiIconButton-label-86"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root-91 MuiIconButton-icon-87"
                    color={undefined}
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                    />
                  </svg>
                </span>
                <span
                  className="MuiTouchRipple-root-98"
                />
              </button>
            </div>
          </div>
        </td>
      </tr>
    </tfoot>
  </table>
</div>
`;
